api/python/provisioner/commands/set_swupgrade_repo.py

```diff
@@ -150,6 +157,40 @@ def _single_repo_validation(self, release, repo_name):
                     )
                 )
 
+    @staticmethod
+    def _validate_python_index(index_path: Path):
+        """
+        Perform the dynamic validation for SW upgrade Python index by
+        the given index path `index_path`
+
+        Parameters
+        ----------
+        index_path: Path
+            Path to the SW upgrade Python index
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        SWUpdateRepoSourceError
+            If Python index validation fails
+
+        """
+        logger.debug("Start Python index validation")
+        test_package_name = next(index_path.iterdir())
+        cmd = (f"pip3 download {test_package_name} --dest=/tmp/ "
```

Comment:
two things:
- /tmp is not good, please use https://docs.python.org/3.6/library/tempfile.html
- need to remove temp objects

Reply:
Yes, sure. It was just a WIP solution, I moved to use `tempfile` this module is self-responsible for tmp directories deleting